keep/providers/opensearchserverless_provider/opensearchserverless_provider.py

```diff
@@ -0,0 +1,209 @@
+            mandatory=True,
+            alias="Read Access",
+        ),
+        ProviderScope(
+            name="aoss:WriteDocument",
+            description="Required to save documents.",
+            documentation_url="https://docs.aws.amazon.com/opensearch-service/latest/developerguide/serverless-genref.html#serverless-operations",
+            mandatory=True,
+            alias="Write Access",
+        ),
+    ]
+
+    def __init__(
+        self, context_manager: ContextManager, provider_id: str, config: ProviderConfig
+    ):
+        self.auth = None
+        self.client = None
+        super().__init__(context_manager, provider_id, config)
+
+    def __get_url(self, paths: List[str] = [], query_params: dict = None, **kwargs):
+        """
+        Helper method to build the url for Opensearch api requests.
+        """
+        host = self.authentication_config.domain_endpoint.rstrip("/").rstrip()
+        self.logger.info(f"Building URL with host: {host}")
+        url = urljoin(
+            host,
+            "/".join(str(path) for path in paths),
+        )
+
+        # add query params
+        if query_params:
+            url = f"{url}?{urlencode(query_params)}"
+
+        return url
+
+    def validate_scopes(self):
+        scopes = {scope.name: False for scope in self.PROVIDER_SCOPES}
+        test_index = "keep-validation-index"
+        test_doc_id = "keep-test-doc"
+        test_doc = {"message": "Keep test doc"}
+
+        # Validate Write
+        try:
+            res = self.__create_doc(test_index, test_doc_id, test_doc)
+            if res.status_code in [200, 201]:
+                scopes["aoss:WriteDocument"] = True
+                self.logger.info("Write permission validated successfully.")
+
+                # Clean up
+                delete_res = self.__delete_doc(test_index, test_doc_id)
+                if delete_res.status_code not in [200, 202]:
+                    self.logger.error(f"Failed to delete test doc: {delete_res.status_code} - {delete_res.text}")
+            elif res.status_code == 403:
+                self.logger.error("No permission for aoss:WriteDocument")
+            else:
+                self.logger.error(f"Unexpected response while testing write: {res.status_code}, {res.text}")
+        except Exception as e:
+            self.logger.error("Error while testing aoss:WriteDocument", extra={"exception": str(e)})
+
+        # Validate Read
+        try:
+            res = self.__get_doc(test_index, test_doc_id)
+            if res.status_code == 403:
+                self.logger.error("No permission for aoss:ReadDocument")
+            else:
+                scopes["aoss:ReadDocument"] = True
+                self.logger.info("Read permission validated successfully.")
+        except Exception as e:
+            self.logger.error("Error while testing aoss:ReadDocument", extra={"exception": str(e)})
+
+        return scopes
```

Comment:
This reminds me of the problem we previously had with OpsGenie where we created a test alert to test scopes.
I think it's a bad practice and we should try hard to see if there's another way to validate the scopes the user has without creating/deleting anything.
For example, one may give you access to create a doc but not to delete it, and we will fail because of that, though we do have the access to create the doc.